### PR TITLE
Remove transport type entity and replace it with enum class

### PR DESCRIPTION
**Pull request by sspooky13**
#### Description, the reason for the PR
Transport type entity is unnecessary because admin can only edit name, so it's better to replace it hard in code

#### Fixes issues
...

#### Check the appropriate checkboxes below, please

- [x] [BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/) 
- [ ] New feature 
- [x] I have read and signed [License Agreement for contributions](https://www.shopsys.com/license-agreement)

## Summary by CodeRabbit

- **New Features**
	- Introduced a streamlined approach for handling transport types, enhancing maintainability and performance.
	- Added constants for transport type codes for improved readability.
	- Added `transportTypeCode` field in various GraphQL mutations and queries for better transport type representation.

- **Bug Fixes**
	- Removed outdated transport type roles and related functionalities to simplify user permissions.

- **Documentation**
	- Updated translation files to reflect the removal of transport type-related entries and added new entries for "Packetery."

- **Chores**
	- Removed several obsolete templates and classes related to transport types, cleaning up the codebase.

----
:globe_with_meridians: Live Preview:
  - https://pt-ssp-2237-transport-type.odin.shopsys.cloud
  - https://cz.pt-ssp-2237-transport-type.odin.shopsys.cloud

### CODE REVIEW

File: packages/framework/src/Form/Admin/Transport/TransportFormType.php [hunk 1]
```diff
@@ -97,11 +95,13 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 'empty_message' => t('You have to create some payment first.'),
                 'label' => t('Available payment methods'),
             ])
-            ->add('transportType', ChoiceType::class, [
+            ->add('type', ChoiceType::class, [
                 'required' => true,
-                'choices' => $this->transportTypeFacade->getAll(),
-                'choice_label' => 'name',
-                'choice_value' => 'id',
+                'choices' => [
```
Comment (vitek-rostislav) [inline packages/framework/src/Form/Admin/Transport/TransportFormType.php]: I would rather create `getAllIndexedByTranslations()` method directly in `TransportTypeEnum`

File: packages/frontend-api/src/Resources/config/graphql-types/ModelType/Transport/TransportDecorator.types.yaml [hunk 2]
```diff
@@ -38,6 +38,6 @@ TransportDecorator:
             payments:
                 type: "[Payment!]!"
                 description: "List of assigned payments"
-            transportType:
-                type: "TransportType!"
-                description: "Type of transport"
+            transportTypeCode:
+                type: "String!"
```
Comment (vitek-rostislav) [inline packages/frontend-api/src/Resources/config/graphql-types/ModelType/Transport/TransportDecorator.types.yaml]: I would rather create the `TransportTypeEnum` + `Decorator` here, see eg https://github.com/shopsys/shopsys/blob/16.0/packages/frontend-api/src/Resources/config/graphql-types/EnumType/LoginTypeEnumDecorator.types.yaml

File: upgrade-notes/backend_20240912_121631.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#### Removed transport type entity and replaced it with enum class ([#3431](https://github.com/shopsys/shopsys/pull/3431))
+
```
Comment (vitek-rostislav) [inline upgrade-notes/backend_20240912_121631.md:2]: the BC breaks should be mentioned here (but if this goes to 16.0, then it is ok :smile: )

**Review by vitek-rostislav — DISMISSED** [review]

File: packages/framework/src/Form/Admin/Transport/TransportFormType.php (hunk 1 above)
Comment (sspooky13) [reply packages/framework/src/Form/Admin/Transport/TransportFormType.php]: 🫡 fixed https://github.com/shopsys/shopsys/pull/3431/commits/a15ed6f59bd41e6d10e0ea4cfdb6e247b1279643

File: packages/frontend-api/src/Resources/config/graphql-types/ModelType/Transport/TransportDecorator.types.yaml (hunk 2 above)
Comment (sspooky13) [reply packages/frontend-api/src/Resources/config/graphql-types/ModelType/Transport/TransportDecorator.types.yaml]: 🫡 fixed https://github.com/shopsys/shopsys/pull/3431/commits/494bdea7928dddc3d677968062b75462f2905b9f and https://github.com/shopsys/shopsys/pull/3431/commits/a1cd5838bd7c69ef0b65a83651843b7e8431a15c

File: upgrade-notes/backend_20240912_121631.md (hunk 3 above)
Comment (sspooky13) [reply upgrade-notes/backend_20240912_121631.md:2]: what exactly should I mention? 🤔 I don't know how to exactly write it 😅

**Review by grossmannmartin — APPROVED** [review]

**Pull request merged**